pkg/apis/cloudcredential/v1/aws_types.go

```diff
@@ -39,10 +40,10 @@ type StatementEntry struct {
 	// Resource specifies the object(s) this statement should apply to. (or "*" for all)
 	Resource string `json:"resource"`
 	// PolicyCondition specifies under which condition StatementEntry will apply
-	PolicyCondition IAMPolicyCondition `json:"policyCondition,omitempty"`
+	PolicyCondition *runtime.RawExtension `json:"policyCondition,omitempty"`
```

Comment:
Does this need to be a pointer? It feels like this could trigger the situation deads warned about where going to a pointer means old clients could suddenly start getting nil's, and break list+watch hard.